packages/components/src/popover/test/index.tsx

```diff
@@ -36,6 +37,25 @@ type PlacementToInitialTranslationTuple = [
 	CSSProperties[ 'translate' ],
 ];
 
+let offsetHeightSpy: jest.SpiedGetter<
+	typeof HTMLElement.prototype.offsetHeight
+>;
+
+beforeAll( () => {
+	offsetHeightSpy = jest
+		.spyOn( HTMLElement.prototype, 'offsetHeight', 'get' )
+		.mockImplementation( function getOffsetHeight( this: HTMLElement ) {
+			if ( this.tagName === 'BODY' ) {
+				return window.outerHeight;
+			}
+			return 50;
+		} );
+} );
+
+afterAll( () => {
+	offsetHeightSpy?.mockRestore();
```

Comment:
Nit: I slightly prefer `jest.restoreAllMocks()` here, because if I see a global `let offsetHeightSpy` being declared there's an extra cognitive step of wondering if there are any test assertions on that specific spy.